folksonomy/api.py

```diff
@@ -841,9 +841,11 @@ async def get_user_roles_from_db(user_id: str):
         'SELECT admin, moderator, "user" FROM auth WHERE user_id = %s', (user_id,)
     )
     result = await cur.fetchone()
-    if result:
-        return {"admin": result[0], "moderator": result[1], "user": result[2]}
-    return {"admin": False, "moderator": False, "user": True}
+    if not result:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="User roles not found"
+        )
```

Comment:
I don't think it's a good idea to raise a 404 here, because it will really depends on the context where we ask those roles.

I think a specific exception derived from RuntimeException would be better.

An simpler acceptable option would be a 401 (authorization required). I will modify to get this for now.

```suggestion
    if not result:
        raise HTTPException(
            status_code=status. HTTP_401_UNAUTHORIZED, detail="User roles not found"
        )
```